src/lambda/trans.lisp

```diff
@@ -36,37 +36,43 @@
           (unit
            (terminal (stlc-ctx-to-mu context)))
           ((left term)
-           (comp (->left (mcar (inf-pass context tterm)) (mcadr (inf-pass context tterm)))
-                 (compile-checked-term context term)))
+           (let ((tott (type-of-term context tterm)))
```

Comment:
I have pretty big issue with this code, it seems to me that we never cache. If we are running this recursively, we will end up calling `type-of-term` which does no caching.

Thus it'd be much better if you only call `ann-term1` once. and in the recursive calls, just get the `ttype`

Reply:
Will be fixed in the next commit